uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -932,6 +944,20 @@ define([
       }
 
       /**
+       * Loads the attachmentss for current leave request
+       *
+       * @return {Promise}
+       */
+      function loadAttachments() {
+        //In CREATE mode dont fetch comments
```

Comment:
I don't like this tbh, same goes for the comments. The controller shouldn't concern itself with whether it's safe to call or not to call `instance.loadAttachments` or `instance.loadComments`. The only reason for why we are doing this check is only because on "create" mode the leave request instance wouldn't have the `id` property, and so the api call would fail.

But again, this is something that needs to be handled within the leave request instance itself, not on the controller. The controller should only call
```js
this.request.loadComments()
```
and then that either returns the array of comments or an empty array, either because it has none, or because the leave request is not on the DB yet